dot4gravity/src/lib.rs

```diff
@@ -1,8 +1,707 @@
+// Ajuna Node
+// Copyright (C) 2022 BlogaTech AG
+
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+use rand::Rng;
+
 #[cfg(test)]
-mod tests {
-    #[test]
-    fn it_works() {
-        let result = 2 + 2;
-        assert_eq!(result, 4);
+mod tests;
+
+const BOARD_WIDTH: usize = 10;
+const BOARD_HEIGHT: usize = 10;
+const NUM_OF_PLAYERS: usize = 2;
+const NUM_OF_BOMBS_PER_PLAYER: usize = 3;
+const NUM_OF_BLOCKS: usize = 10;
+
+type Player = u32;
+
+/// Represents a cell of the board.
+#[derive(Clone, Copy, Debug, Eq, PartialEq)]
+enum Cell {
+    Empty,
+    Bomb([Option<Player>; NUM_OF_PLAYERS]),
+    Block,
+    Stone(Player),
+}
+
+impl Cell {
+    /// Tells if a cell is suitable for dropping a bomb.
+    fn is_valid_for_dropping_bomb(&self) -> bool {
+        match self {
+            Cell::Empty => true,
+            Cell::Bomb([Some(_), None]) => true,
+            _ => false,
+        }
```

Comment:
Is the logic correct that we can drop bombs on a cell with a bomb already?

And can we move this function below `is_bomb` and reuse it:
```suggestion
        *self == Cell::Empty || self.is_bomb()
```

Reply:
Yes, a cell can hold bombs from different players but no more than 1 bomb from the same player.